Update server 4.3+ docs on DLC requiring S3 bucket versioning [ONPREM-1361] (#8870)
circleci/circleci-docs: jekyll/_cci2/server/v4.3/installation/phase-1-prerequisites.adoc

    --create-bucket-configuration LocationConstraint=<YOUR_REGION>
----

==== b. Enable bucket versioning

> To use the https://circleci.com/docs/docker-layer-caching/[Docker layer caching] (DLC) feature in CircleCI, https://docs.aws.amazon.com/AmazonS3/latest/userguide/Versioning.html[bucket versioning] needs to be enabled. Run the following command to do that on the bucket created in the previous step:

Reviewer: ```suggestion
To use the link:https://circleci.com/docs/docker-layer-caching/[Docker layer caching] (DLC) feature in CircleCI, link:https://docs.aws.amazon.com/AmazonS3/latest/userguide/Versioning.html[bucket versioning] needs to be enabled. Run the following command to enable bucket versioning on the bucket created in the previous step:
```
